src/UI/EditorLayoutView.re

```diff
@@ -45,40 +35,30 @@ let renderTree = (state, theme, tree) => {
            height(item.height),
          ]>
          {switch (
-            EditorGroups.getEditorGroupById(
-              state.editorGroups,
-              item.split.editorGroupId,
-            )
+            EditorGroups.getEditorGroupById(state.editorGroups, item.content)
           ) {
-          | Some(editorGroup) =>
-            <EditorGroupView
-              state
-              theme
-              windowId={item.split.id}
-              editorGroup
-            />
+          | Some(editorGroup) => <EditorGroupView state theme editorGroup />
           | None => React.empty
           }}
        </View>
      )
   |> React.listToElement;
 };
 
-let make = (~state: State.t, ~theme, ()) => {
-  let {State.windowManager, _} = state;
-  let {windowTree, _} = windowManager;
-
-  let children = renderTree(state, theme, windowTree);
+let%component make = (~state: State.t, ~theme, ()) => {
+  let%hook (maybeDimensions, setDimensions) = Hooks.state(None);
+  let children =
+    switch (maybeDimensions) {
+    | Some((width, height)) =>
+      renderTree(~width, ~height, state, theme, state.layout)
+    | None => React.empty
+    };
```

Comment:
Instead of sending the dimensions all the way to the `WindowManager` state and back, it's now contained here with a state hook.

Reply:
Cool!